fix(backend): gate AI exclusively by settings.platform_ai_enabled (#13568)

amiraifori opened the pull request:
Fixes #13568

- Remove legacy ai:enabled gating from AI LLM initialization (ai-llm.ts)
- Add runtime guard in ai-domain.ts to block AI operations when settings.platform_ai_enabled is false
- Add integration coverage ensuring AI mutations are blocked when platform AI is disabled

Note: CI/integration tests were not run locally due to a Windows test bootstrap crash referencing detectors/platform/node/machine-id/execAsync (unrelated to this change).

aHenryJard commented on opencti-platform/opencti-graphql/src/database/ai-llm.ts:
```diff
@@ -27,7 +26,7 @@ const AI_AZURE_DEPLOYMENT = conf.get('ai:ai_azure_deployment');
 let client: Mistral | OpenAI | AzureOpenAI | null = null;
 let nlqChat: ChatOpenAI | ChatMistralAI | AzureChatOpenAI | null = null;
-if (AI_ENABLED && AI_TOKEN) {
+if (AI_TOKEN) {
```
I think we need to keep the AI_ENABLED but get the value from settings instead of conf.get('ai:enabled');

amiraifori commented:
Hello @aHenryJard
Hope you are doing alright, just a quick reminder I would appreciate it if my PR is reviewed.

aHenryJard commented on opencti-platform/opencti-graphql/src/database/ai-llm.ts:
```diff
@@ -99,12 +111,29 @@ if (AI_ENABLED && AI_TOKEN) {
+    resetClients();
+    return;
+  }
+  initClients();
```
I think this will init client at each AI call instead of once as it was before.

Would be better to init client only if client is null, and reset only if client is not null and AI is switched to disabled.

Could you try that way please ?

aHenryJard reviewed (commented):
Would be better to init client only if client is null, and reset only if client is not null and AI is switched to disabled.

I was thinking on init/reset on settings edit field but it will not work on clusters. So the best way that I see to have it dynamically taken is the one you have implemented, but we should find a way to init only once and not at each AI calls.

amiraifori commented:
Hello @aHenryJard 
Hope you are doing alright, just a quick reminder I would appreciate it if my PR is reviewed.

amiraifori commented:
Hello @aHenryJard , @SamuelHassine 
Hope you are doing alright, just a quick reminder I would appreciate it if my PR is reviewed.

amiraifori committed:
fix(backend): gate AI exclusively by settings.platform_ai_enabled (#13568)

Gate AI by both ai:enabled config and settings.platform_ai_enabled, centralize the disabled error message, keep client init lazy/idempotent (init once, reset only on state change), and add integration coverage for both the AI module mutation (aiFixSpelling) and a domain-level AI entry point (stixCoreObjectAskAiActivity).

Signed-off-by: amiraifori <iforiamira@gmail.com>

aHenryJard commented:
Hello, all AI configuration is going to be removed in OpenCTI in favor of xtm-one, I don't think it's worthy to continue this pull request so I'm closing.
